docs/design/HealthEndpoint.md

```diff
@@ -0,0 +1,206 @@
+```
+{
+    "status": "Healthy",
+    "version": "Major.Minor.Patch",
+    "appName": "dab_oss_Major.Minor.Patch"
+}
+```
+
+This is generated in the `Startup.cs` file when mapping the base URL `(/)` to the health check endpoint to get the high level result for DAB Engine Health using this command `ResponseWriter = app.ApplicationServices.GetRequiredService<HealthReportResponseWriter>().WriteResponse`.
+
+| **Term**           | **Description**                                         |
+|--------------------|---------------------------------------------------------|
+| **Health Endpoint** | The URL (e.g., `/`) exposed as JSON.             |
+| **Check**           | A specific diagnostic test (e.g., database, API).      |
+| **Status**          | The result of a check.                                 |
+| **Status.Healthy**  | The system is functioning correctly.                   |
+| **Status.Unhealthy**| The system has a critical failure or issue.            |
+
+## Proposed Change
+We want to create a detailed version of the DAB Engine Health endpoint with information regarding all REST and GraphQL endpoints and their behaivour. Here we would update the configuration schema to include what all results or checks do we want to perform on that runtime/data-source/entity (described below).
+We want the user to provide a check and threshold value what DAB would carry out and then check if the result is under the threshold, if so, the DAB Engine would be considered healthy for that endpoint, else unhealthy. 
+
+> For Example, if the data source health is being performed and the user mentions that we would do a GET of all resources from the DB and this should be under 25sec. This check would be given as a SQL Query that the users wants to execute on the DB, then after performing this check, we will validate if the condition is satisfied, else this data source moniker would be considered unhealthy.
+
+
+## Implementation Details
+### Schema
+Health check responses follow a common convention rather than a strict standard. The typical pattern involves a "checks" property for individual components' statuses (e.g., database, memory), with each status rolling up to an overall "status" at the top level.
+```
+{
+  "status": "Healthy",
+  "checks": {
+    "check-name": { "status": "Healthy" },
+    "check-name": { "status": "Healthy" }
+  }
+}
+```
+
+### Configuration Updates
+We need to update the dab config file to include details of the health check for different configuration parameters like runtime, data source, entities. 
+
+#### `runtime.health` Configuration
+
+The runtime configuration would include details like cache-ttl in case we need to cache the response of health checks, the max-drop value which tells you the degree of parallelism i.e. how many queries that DAB shoud run at once to get health results and roles i.e. which role is allowed to view the health information of DAB.
```

Comment:
```suggestion
The runtime configuration would include details like cache-ttl in case we need to cache the response of health checks, the max-dop value which tells you the degree of parallelism i.e. how many queries that DAB shoud run at once to get health results and roles i.e. which role is allowed to view the health information of DAB.
```